source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.Application/MeteringPoints/Handlers/ConsumptionMeteringPointPersister.cs

```diff
@@ -40,8 +40,39 @@ public async Task PersistAsync(ConsumptionMeteringPointCreatedEvent consumptionM
                 throw new ArgumentNullException(nameof(consumptionMeteringPointCreatedEvent));
 
             var meteringPoint = MeteringPointFactory.Create(consumptionMeteringPointCreatedEvent);
-            await _meteringPointRepository.StoreMeteringPointAsync(meteringPoint).ConfigureAwait(false);
-            _logger.LogInformation("Finished persisting metering point with id: {meteringPointId}", meteringPoint.MeteringPointId);
+
+            var existingMeteringPoint = await _meteringPointRepository.GetOrNullAsync(meteringPoint.MeteringPointId);
+
+            if (existingMeteringPoint == null)
+            {
+                await _meteringPointRepository.StoreMeteringPointAsync(meteringPoint).ConfigureAwait(false);
+                _logger.LogInformation($"Consumption Metering Point ID '{meteringPoint.MeteringPointId}' has been persisted");
+            }
+            else
+            {
+                _logger.LogInformation(
+                    $"Metering Point ID '{meteringPoint.MeteringPointId}' already exists in storage");
+
+                // Compare and log differences between the integration event data and the persisted metering point's data
+                CompareMeteringPoints(meteringPoint, existingMeteringPoint);
```

Comment:
It's hard to tell what's going on here and why without looking at the implementation. Perhaps?
```suggestion
                ValidateMeteringPointsIdenticalOrThrow(meteringPoint, existingMeteringPoint);
```